ProcessLib/Parameter/Parameter.h

```diff
@@ -13,18 +13,14 @@
 #include <memory>
 #include <utility>
 #include <vector>
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
 #include "SpatialPosition.h"
 
 namespace BaseLib
 {
 class ConfigTree;
 }  // BaseLib
 
-namespace MathLib
-{
-class PiecewiseLinearInterpolation;
-}  // MathLib
```

Comment:
Hm. IMHO the forward declaration should suffice here. But then maybe the include has to be added to `Parameter.cpp`.

Reply:
Unfortunately not, because the unique_ptr needs the definition of the destructor.

Reply:
Even though it's passed as const reference? Strange. Another thing learned.

Reply:
See [SO](https://stackoverflow.com/a/6089065) and the [reference](http://howardhinnant.github.io/incomplete.html) therein.
[cppreference](http://en.cppreference.com/w/cpp/memory/unique_ptr) (search "incomplete type") also has a paragraph about the incomplete type requirements of the unique_ptr, but more difficult to understand.

Reply:
Hmmm, now it's working :confused: